git_hooks/pre-push

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+# Called by "git push" after it has checked the remote status, but before anything has been
+# pushed.  If this script exits with a non-zero status nothing will be pushed.
+#
+# This hook is called with the following parameters:
+#
+# $1 -- Name of the remote to which the push is being done
+# $2 -- URL to which the push is being done
+#
+# If pushing without using a named remote those arguments will be equal.
+#
+# Information about the commits which are being pushed is supplied as lines to
+# the standard input in the form:
+#
+#   <local ref> <local sha1> <remote ref> <remote sha1>
+#
+
+remote="$1"
+url="$2"
+
+echo "remote is $remote"
+echo "url is $url"
+
+./tests/scripts/check-doxy-blocks.pl
+if [ $? != 0 ]
```

Comment:
once I added set -eu, no need to check return statement, as the script will exit once the scripts fail